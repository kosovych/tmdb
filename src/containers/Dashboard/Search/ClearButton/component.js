import React from 'react'
import {
  CloseCircleOutlined
} from '@ant-design/icons'
import PropTypes from 'prop-types'
import classnames from 'classnames'

const ClearButton = ({ onClick, isVisible }) => (
  <button
    type="button"
    onClick={onClick}
    className={classnames('reset-btn', { 'd-none': !isVisible })}
  >
    <CloseCircleOutlined />
  </button>
)

ClearButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  isVisible: PropTypes.bool.isRequired
}

export default ClearButton
