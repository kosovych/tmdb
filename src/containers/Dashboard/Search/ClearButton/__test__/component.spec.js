import React from 'react'
import { shallow } from 'enzyme'
import ClearButton from '../component'

describe('ClearButton component', () => {
  const props = {
    onClick: jest.fn(),
    isVisible: true
  }
  it('should match snapshot', () => {
    const component = shallow(<ClearButton {...props} />)
    expect(component).toMatchSnapshot()
  })
})
