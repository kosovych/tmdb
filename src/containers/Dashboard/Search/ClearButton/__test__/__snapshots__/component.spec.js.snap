// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton component should match snapshot 1`] = `
<button
  className="reset-btn"
  onClick={[MockFunction]}
  type="button"
>
  <ForwardRef(CloseCircleOutlined) />
</button>
`;
